manifests/profile/master/install.pp

```diff
@@ -5,19 +5,28 @@
 # @example Apply this class to the Master and any/all Compilers
 #   include puppet_metrics_dashboard::profile::master::install
 #
-class puppet_metrics_dashboard::profile::master::install {
+class puppet_metrics_dashboard::profile::master::install (
+  Boolean $manage_ldap_auth = true,
+  ) {
+
   if $facts['pe_server_version'] {
     $puppetserver_service = 'pe-puppetserver'
   } else {
     $puppetserver_service = 'puppetserver'
   }
 
-  if $puppet_metrics_dashboard::enable_ldap_auth {
+  if manage_ldap_auth {
     package { 'toml':
       ensure   => present,
       provider => 'puppetserver_gem',
       notify   => Service[$puppetserver_service],
     }
+  } else {
+    package { 'toml':
+      ensure   => absent,
```

Comment:
The `manage_ldap_auth` name indicates that we will not manage the gem if that is unset. The reason for that logic is so that a user can disable the management of the gem if they have it declared elsewhere in their code. With this current implementation, they will not be able to disable the management of it.

As a coding note, if this is to be used as an `ensure`, you can just drop the `if/else` and just do.

```
package { 'toml':      
  ensure   =>   $manage_ldap_auth
```

Or something like

```
$toml_ensure = ? $install_toml {
  true => 'present'
  default => 'absent'
}

package { 'toml':
  ensure => $install_toml
```

However, I would probably just remove the `else` part of this since the other package doesn't have this same logic.

Reply:
Sounds good! I'll get that changed and put together a spec test.